Sources/SmokeDynamoDB/InMemoryDynamoDBCompositePrimaryKeyTableWithIndex.swift

```diff
@@ -27,7 +27,42 @@ public enum GSIError: Error {
 
 public struct InMemoryDynamoDBCompositePrimaryKeyTableWithIndex<GSILogic: DynamoDBCompositePrimaryKeyGSILogic>: DynamoDBCompositePrimaryKeyTable {
     public func monomorphicBulkWriteWithoutThrowing<AttributesType, ItemType>(_ entries: [WriteEntry<AttributesType, ItemType>]) async throws -> [Int : BatchStatementError] where AttributesType : PrimaryKeyAttributes, ItemType : Decodable, ItemType : Encodable {
-        return try await self.primaryTable.monomorphicBulkWriteWithoutThrowing(entries)
+        var errors: [Int: BatchStatementError] = [:]
```

Comment:
You are modifying this dictionary across different EventLoopFutures. This probably is ok as they should all be running on the same thread but this is not guaranteed. I would use the result of the EventLoopFuture - returning something like a (String, BatchStatementError)? - to construct the errors dictionary after all the EventLoopFutures have completed.

Reply:
thanks for the feedback. updated!